### DISCUSSION TITLE
Add support for scanner mods

### DISCUSSION THREAD

**Discussion by Test-Account666** [Malcolm-Q/LC-LateGameUpgrades#529] [General]
Hey!

There is an incompatible scanner mod and one upcoming scanner mod that will be incompatible.

They are called "GoodItemScan" and "QuickItemScan" respectively.

GoodItemScan already has an API for LGU to use.

It would be nice to see support for them :)

**Comment by WhiteSpike** [comment]
If you could describe what is the issue exactly, sure.

Otherwise, I'm clueless where I'm supposed to use the API on. :D

**Reply by Test-Account666** [reply]
> If you could describe what is the issue exactly, sure.
> 
> Otherwise, I'm clueless where I'm supposed to use the API on. :D

Well, there's the upgrade that allows for scanning through walls.

There's also the extra range one.

Both do not work.

There's a dedicated class called "CheatsAPI" which allows mods to adjust those things

**Reply by WhiteSpike** [reply]
What am I supposed to put when the no line of sight is a toggle? Just a big number?

**Reply by mattymatty97** [reply]
tbh both of the unlocks can be done using vanilla code:
scan though walls by setting the `requiresLineOfSight` field of a scanNodeProperties to false
and the extended range by increasing the `maxRange` of a scanNodeProperties

**Reply by mattymatty97** [reply]
TL;DR: if you were to change the properties on the scanNodeProperties instead of patching the HUDManager code you would be compatible with virtually any scanner mod

**Reply by WhiteSpike** [reply]
Wouldn't that mean I would have to change every single properties that spawns whenever the upgrade is bought?

**Reply by Test-Account666** [reply]
> Wouldn't that mean I would have to change every single properties that spawns whenever the upgrade is bought?

Yes, that's exactly what it would mean...



> What am I supposed to put when the no line of sight is a toggle? Just a big number?

You can just use a big number, yeah. I can also add another variable, if you prefer that

**Comment by Test-Account666** [comment]
Any updates on this? :)

**Reply by WhiteSpike** [reply]
I already released an update which uses the API.
If you want to use the scan node property route, you will have to make a PR with it as I'm currently focused on thesis for masters degree rather than modding.
